src/app/EventManagement.cpp

```diff
@@ -748,6 +751,71 @@ CHIP_ERROR EventManagement::FetchEventsSince(TLVWriter & aWriter, const ObjectLi
     return err;
 }
 
+CHIP_ERROR EventManagement::RemoveInvalidFabricCB(const TLV::TLVReader & aReader, size_t aDepth, void * apContext)
+{
+    // the function does not actually remove the event, instead, it sets the fabric index to an invalid value.
+    FabricIndex * invalidFabricIndex = static_cast<FabricIndex *>(apContext);
+
+    TLVReader event;
+    TLVType tlvType;
+    TLVType tlvType1;
+    event.Init(aReader);
+    ReturnErrorOnFailure(event.EnterContainer(tlvType));
```

Comment:
Do we really want to fail out of the overall iteration in these various failure cases, or just move on to the next item?

Reply:
preferring moving on to the next item

Reply:
done, thanks